collector/hwmon_linux.go

```diff
@@ -47,13 +60,25 @@ func init() {
 }
 
 type hwMonCollector struct {
-	logger log.Logger
+	hwmonIncludePattern *regexp.Regexp
+	hwmonExcludePattern *regexp.Regexp
+	logger              log.Logger
 }
 
 // NewHwMonCollector returns a new Collector exposing /sys/class/hwmon stats
 // (similar to lm-sensors).
 func NewHwMonCollector(logger log.Logger) (Collector, error) {
-	return &hwMonCollector{logger}, nil
+
+	level.Info(logger).Log("msg", "Parsed flag --collector.hwmon.unit-include", "flag", *hwmonInclude)
+	hwmonIncludePattern := regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *hwmonInclude))
+	level.Info(logger).Log("msg", "Parsed flag --collector.hwmon.unit-exclude", "flag", *hwmonExclude)
+	hwmonExcludePattern := regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *hwmonExclude))
```

Comment:
Here if a bad regexp is provided, I think the program will panic due to `MustCompile` behavior, which may not be user-friendly given we can report the error with the logger? Or given the `error` in the function signature it may be okay to just pass the errors up one level too.

Reply:
Good catch!

Reply:
Updated this PR to use device_filter.go, as demonstrated in https://github.com/prometheus/node_exporter/pull/2432